app/javascript/utilities/dropdownUtils.js

```diff
@@ -177,15 +148,17 @@ export const initializeDropdown = ({
       closeDropdown({
         triggerElementId,
         dropdownContentId,
-        onClose,
+        onClose: onCloseCleanupActions,
       });
     } else {
       openDropdown({
         triggerElementId,
         dropdownContentId,
-        onClose,
       });
       onOpen?.();
+
+      document.addEventListener('keyup', keyUpListener);
+      document.addEventListener('click', clickOutsideListener);
```

Comment:
Refactor (non-blocking): Consider only adding the event listener once so it does not need to be cleaned up in

```javascript
  const onCloseCleanupActions = () => {
    onClose?.();
    document.removeEventListener('keyup', keyUpListener);
    document.removeEventListener('click', clickOutsideListener);
  };
```


```suggestion
      document.addEventListener('click', clickOutsideListener, { once: true ));
```

Reply:
I was looking at this - initially I thought it might be ideal but sometimes we need the event listener to fire more than once - e.g. if a user clicks inside of the dropdown content I think `{ once: true }` would remove the listener, but we'd want it to stay attached 🤔

Reply:
Ahh, good point. 😎

Reply:
I really wanted to use it 😂  😭